src/pages/fi-dimensions/index.js

```diff
@@ -0,0 +1,226 @@
+
+  const [tempDimCount, setTempDimCount] = useState(null)
+
+  const [initialValues, setInitialData] = useState({
+    DimCount: null,
+    tpaDimension1: null,
+    tpaDimension2: null,
+    tpaDimension3: null,
+    tpaDimension4: null,
+    tpaDimension5: null,
+    tpaDimension6: null,
+    tpaDimension7: null,
+    tpaDimension8: null,
+    tpaDimension9: null,
+    tpaDimension10: null,
+    tpaDimension11: null,
+    tpaDimension12: null,
+    tpaDimension13: null,
+    tpaDimension14: null,
+    tpaDimension15: null,
+    tpaDimension16: null,
+    tpaDimension17: null,
+    tpaDimension18: null,
+    tpaDimension19: null,
+    tpaDimension20: null
+  })
+
+  const { formik } = useForm({
+    initialValues: initialValues,
+    enableReinitialize: true,
+    validateOnChange: true,
+
+    validationSchema: yup.object({
+      DimCount: yup.number().nullable().required('Dim Count is required').min(1).max(20),
+      ...Object.fromEntries(
+        Array.from({ length: 20 }, (_, i) => i + 1).map(num => [
+          `tpaDimension${num}`,
+          yup
+            .string()
+            .nullable()
+            .test(`is-tpaDimension${num}-required`, `Dimension ${num} is required`, function (value) {
+              const { DimCount } = this.parent
+
+              return DimCount >= num ? value != null : true
+            })
+        ])
+      )
+    }),
+
+    onSubmit: values => {
+      postDimensionSettings(values)
+    }
+  })
+
+  useEffect(() => {
+    getDataResult()
+  }, [])
+
+  const getDataResult = () => {
+    const myObject = {}
+    var parameters = `_filter=`
+    getRequest({
+      extension: SystemRepository.Defaults.qry,
+      parameters: parameters
+    })
+      .then(res => {
+        const filteredList = res.list.filter(obj => {
+          return Object.keys(initialValues).includes(obj.key)
+        })
+
+        filteredList.forEach(obj => {
+          if (obj.value && !isNaN(obj.value) && obj.value.trim() !== '') {
+            myObject[obj.key] = parseInt(obj.value)
+          } else {
+            myObject[obj.key] = obj.value
+          }
+        })
+
+        formik.setValues(myObject)
+        console.log('obj', myObject)
+      })
+      .catch(error => {})
+  }
+
+  const { labels: _labels, access } = useResourceQuery({
+    datasetId: ResourceIds.FI_dimensions
+  })
+
+  const postDimensionSettings = obj => {
+    var dataToPost = [{ key: 'DimCount', value: obj.DimCount }]
+    for (let i = 1; i <= 20; i++) {
+      const dimKey = `tpaDimension${i}`
+      if (obj[dimKey] !== undefined) {
+        dataToPost.push({ key: dimKey, value: obj[dimKey] })
+      }
+    }
+    console.log(dataToPost)
+    postRequest({
+      extension: SystemRepository.Defaults.set,
+      record: JSON.stringify({ sysDefaults: dataToPost })
+    })
+      .then(res => {
+        toast.success('Record Successfully Updated')
+      })
+      .catch(error => {})
+  }
+
+  const handleSubmit = () => {
```

Comment:
is there a need to this function? can't we just call formik.handleSubmit() where we need it directly?